### ISSUE TITLE
Option for replacing sun with LUX sensor

### ISSUE THREAD

**Issue by atterdal** [claytonjn/hass-circadian_lighting#43]
Would it be possible to replace the Sun for a Luminosity sensor for a more accurate interior brightness calculation?

**Comment by claytonjn** [comment]
It's certainly possible, however that would remove the consistency of the lighting on days of bad weather, etc. Consistency is an important part of maintaining your internal Circadian Rhythm so it could negatively impact the effects of adjusting lights automatically. 

Also, my personal experience with lux sensors isn't that great (not smooth transitions, almost acting like a binary bright/not bright sensor):
![image](https://user-images.githubusercontent.com/3850252/65516198-8fd09c80-deae-11e9-952a-8e9fd5b973ef.png)

That being said, brightness shouldn't have as much an effect as color temperature and I'm sure there are better lux sensors out there so I'll mark this as a feature request for me to tackle when I get time.

**Comment by atterdal** [comment]
Agreed that it would create an inconsistency, I think my original idea was more of a daylight mimic/harvest concept during fall/winter here in the North when the sun curve is weird 😊

Cheers!

**Comment by claytonjn** [comment]
Ultimately I need a formula that take in sun altitude and spits out a color temp/brightness. There's been some research correlating sun altitude and CT but I haven't had the time to really comprehend the formulas so they can be translated into python.

**Comment by Joshfindit** [comment]
@claytonjn If you’re looking to do the calc anyway, you might as well go all-out and calculate the frequencies of light that make it to your gps coordinates (what makes it through the atmosphere + what is produced through Raleigh scattering). That would be more accurate than colour temp, and some lights may even be able to reproduce them more accurately.

**Comment by claytonjn** [comment]
> @claytonjn If you’re looking to do the calc anyway, you might as well go all-out and calculate the frequencies of light that make it to your gps coordinates (what makes it through the atmosphere + what is produced through Raleigh scattering). That would be more accurate than colour temp, and some lights may even be able to reproduce them more accurately.

I welcome any contribution in that regard.

**Comment by jferran** [comment]
Hi. First of all thank you for your nice work. I wish I could collaborate, but unfortunately my coding skills are not so good (still struggle with Hass).

In my case years ago I used Domoticz with this script for Circadian lighting: https://www.domoticz.com/forum/viewtopic.php?t=19220

It works in a different way which I think it's better / more accurate to mimic the natural light outside because it checks for the weather (amount of clouds in Okta).

When I used that (in Spain) I had the feeling that there were extra windows at home and that helped me to feel better.

Of course I understand that your approach is different because you want to create a stable rhythm of light independent of the weather.

Once again thank you for job.

**Comment by nightcat91** [comment]
+1 for lux sensor.
Maybe only as an alternative for brightness adjustments, so ct could still be set by sun elevation. 
My lux sensor is pretty accurate at least until it gets dark outside, then it's quickly just 0...
![Screenshot_20191227_000650_com android chrome](https://user-images.githubusercontent.com/31433650/71493186-31181780-283d-11ea-95d4-28121503e6f2.png)

**Comment by SkateWarp** [comment]
+1 for the sensor! :)
